hooks/boot/02-message-helpers.rb

```diff
@@ -39,6 +39,19 @@ def dev_new_install_message(kafo)
 
   def failure_message
     say "\n  <%= color('There were errors detected during install.', :bad) %>"
+
+    # Kafo 7.1.0 added puppet_report
+    if respond_to?(:puppet_report) && puppet_report&.failed_resources&.any?
+      say
+      puppet_report.failed_resources.each do |failed_resource|
+        say "#{failed_resource} failed. Logs:"
+        failed_resource.log_messages.each do |message|
+          say message
+        end
+      end
+      say
+    end
```

Comment:
I'd modify the output more to the following structure. Some blank lines will help with seeing the "chunks" of data. And I would invert the indenting structure so it makes it easier to understand the "section" and then the data within that section. That is indenting the error put but left justifying the hard-coded words themselves.

So that it goes:

<Installer output>
2023-08-24 16:28:49 [NOTICE] [configure] System configuration has finished.

The were errors detected during configuration:

Resource failed. Logs:
  <Error output>

Resource failed. Logs:
  <Error output>

There were errors detected during configuration.
Please address the errors and re-run the installer to ensure the system is properly configured.
Failing to do so is likely to result in broken functionality.

The full log is at /var/log/foreman-installer/foreman.log

Reply:
I've taken the liberty to add code tags to your message so it's clear what you intended.

> Some blank lines will help with seeing the "chunks" of data

Yes, it turns out that `say ""` doesn't emit an empty line which I assumed.

> That is indenting the error put but left justifying the hard-coded words themselves.

Agreed, visual separation isn't there right now. Will update.